### PR TITLE
Add Note entity and Elementable behaviour

### PR DESCRIPTION
**Pull request by ivanovaleksey**
Hi @palkan,

I would like to implement Note entity.
It behaves very similar to existing Task entity, that is why I decided to extract common functionality to `Elementable` module.

Current API [says](https://developers.amocrm.ru/rest_api/tasks_set.php) that Task's `element_type` can be one of the digits in range of 1-3 (and Note's `element_type` too). So I updated it as well.

P. S.
After submitting PR I noticed that #17 implements Note entity too. I think #17 can be closed now.

### CODE REVIEW

**Comment by ivanovaleksey** [comment]
But it is said [here](https://developers.amocrm.ru/rest_api/notes_list.php) under **Response Field Guide** section
> element_type | Тип привязанного элемента: 1 - контакт, 2 - сделка, 3 - компания, 4 - задача (результат задачи)

Is seems line a note can be created for a task 🤔

File: lib/amorail/entities/elementable.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+module Amorail
+  module Elementable
+    extend ActiveSupport::Concern
+
+    ELEMENT_TYPES = [
```
Comment (palkan) [inline lib/amorail/entities/elementable.rb]: Why not just:

```ruby
ELEMENT_TYPES = {
  'contact' => 1,
   ...
}
```

Then we can use it in validation and avoid duplication:

```ruby
validates :element_type, inclusion: ELEMENT_TYPES.values
```

**Review by palkan — CHANGES_REQUESTED** [review]

File: lib/amorail/entities/elementable.rb (hunk 1 above)
Comment (ivanovaleksey) [reply lib/amorail/entities/elementable.rb]: To be honest I just copy-pasted existent solution in constant.
If the structure doesn't matter I totally agree with you.

Comment (palkan) [reply lib/amorail/entities/elementable.rb]: Got it. That was a bad decision, let's make it right!

**Comment by ivanovaleksey** [comment]
@palkan what about my comment above? Having common `Elementable` behaviour seems very delicious, but if Note really can have `element_type = 4 (задача (результат задачи))` it seems to break everything 😞

**Comment by palkan** [comment]
We can extract validation to specific classes, and use smth like`ELEMENT_TYPES.reject { |k, v| k == 'task' }.values` for Task.

**Comment by ivanovaleksey** [comment]
@palkan I have pushed the updates. Please check it once again.
By the way I have used symbols, I think it doesn't matter here since we use constant only to generate methods.

**Review by palkan — APPROVED** [review]

**Pull request merged**